frontend/src/components/AuthForms.tsx

```diff
@@ -0,0 +1,286 @@
+import { Button } from "@/components/ui/button";
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs";
+import { loginSchema, signupSchema } from "@/schemas/authSchemas";
+import postsService from "@/services/postsService";
+import { useAuthStore } from "@/stores/authStore";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useMutation } from "@tanstack/react-query";
+import { AxiosError } from "axios";
+import { UUID } from "crypto";
+import { Eye, EyeOff } from "lucide-react";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+
+type LoginFormData = {
+  username: string;
+  pass: string;
+};
+
+type SignupFormData = {
+  username: string;
+  email: string;
+  pass: string;
+};
+
+interface ErrorResponse {
+  statusCode: number;
+  message: string;
+}
+
+export default function AuthForms() {
+  const [showLoginPassword, setShowLoginPassword] = useState(false);
+  const [showSignupPassword, setShowSignupPassword] = useState(false);
+
+  const [loginError, setLoginError] = useState<string | null>(null);
+  const [signupError, setSignupError] = useState<string | null>(null);
+
+  const navigate = useNavigate();
+
+  const loginMutation = useMutation({
+    mutationFn: async (data: LoginFormData) => {
+      const response: { access_token: string; userId: UUID } =
+        await postsService.logIn(data);
+      return response;
+    },
+    onSuccess: async (response) => {
+      const { access_token, userId } = response;
+
+      useAuthStore.getState().login(access_token, userId);
+
+      const userDetails = await postsService.getUserDetails(
+        userId,
+        access_token
+      );
+
+      const { name, email } = userDetails;
+      useAuthStore.getState().setUserDetails(name, email);
+
+      console.log("Login successful");
+
+      resetLoginForm();
+      navigate("/");
+    },
+    onError: (error: AxiosError<ErrorResponse>) => {
+      setLoginError(error.message);
+    },
+  });
```

Comment:
Create a separate services folder or API folder and put mutations and queries there. I remember I was showing you a cool structure when you create an async function and a hook (useLoginMutation) in which you just return useMutation with the async function called. Put both - function and hook in the same file